test/integration_test.py

```diff
@@ -2062,17 +2070,30 @@ def test_response_security_headers(self):
                     else:
                         response = requests.get(str(endpoint / path))
                         response.raise_for_status()
-                        expected = AzulChaliceApp.security_headers | expected_headers
-                        # FIXME: Add a CSP header with a nonce value to text/html responses
-                        #        https://github.com/DataBiosphere/azul-private/issues/6
+                        expected = AzulChaliceApp.security_headers() | expected_headers
+                        # These paths have a CSP that differs from the default
+                        # by including randomly generated 'nonce' values.
                         if path in ['/', '/oauth2_redirect']:
-                            del expected['Content-Security-Policy']
+                            nonce = set()
+                            for directive in response.headers['Content-Security-Policy'].split(';'):
+                                match = directive_re.fullmatch(directive)
+                                self.assertIsNotNone(match)
+                                name, value = match.groups()
+                                if name in ('script-src', 'style-src'):
+                                    match = nonce_re.search(value)
+                                    self.assertIsNotNone(match)
+                                    nonce.add(match.group(1))
+                            nonce = one(nonce)
+                            # Our nonce token is 32 bytes Base64 encoded
+                            self.assertEqual(43, len(nonce))
+                            expected_csp = AzulChaliceApp.content_security_policy(nonce)
+                            expected['Content-Security-Policy'] = expected_csp
```

Comment:
What is this for?

Reply:
With the nonce token parsed out of the response, it is possible to generate the full expected CSP and add that to the expected headers.